### DISCUSSION TITLE
mailer.Message address fields should be multi-value

### DISCUSSION THREAD

**Discussion by cweagans** [pocketbase/pocketbase#1728] [Q&A]
Per RFC 5322, To, BCC, and CC fields should be a list of addresses (https://datatracker.ietf.org/doc/html/rfc5322#section-3.6.3).

Currently, they are defined like so:

```go
type Message struct {
	[...]
	To          mail.Address
	Bcc         []string
	Cc          []string
	[...]
}
```

All three of those fields should be `[]mail.Address` instead, I think.

**Comment by ganigeorgiev** [comment]
Hm, you are right, but this will be a breaking change and I'm not sure if it is worth applying it because the underlying smtp library [mailyak](https://github.com/domodwyer/mailyak) supports only `[]string` as Cc and Bcc values (_technically To also should be []string but it is a remain from earlier implementation_).

For the next release I'll think a little more on whether to normalize all fields to `[]mail.Address` and internally to pass the serialized version (eg. `name <email>`) as to/bcc/cc string(s) (_I'll have to test it if it will work_).

**Reply by cweagans** [reply]
I suppose I don't have any strong feelings about the specifics -- just that they are all multi-value. It helps a lot for presenting a coherent API.

> this will be a breaking change

I thought the 0.x releases didn't have any BC guarantees? This seems like the time to make these sorts of changes.

**Reply by ganigeorgiev** [reply]
_I've opened an issue in the upstream library (https://github.com/domodwyer/mailyak/issues/74) and based on its outcome (there is also limitation in the std net/smtp package), we'll decide whether to change all fields to `[]mail.Address` or `[]string` (aka. leaving it without the recipient)._

**Reply by ganigeorgiev** [reply]
The upstream library merged the related PR and all `mail.Message` recipient fields (`To`, `Bcc`, `Cc`) were normalized to `[]mail.Address` lists. The change currently is only in the `develop` branch and will be available with the next v0.13 release.
_If you are sending custom emails this will be a minor breaking change (see [CHANGELOG](https://github.com/pocketbase/pocketbase/blob/develop/CHANGELOG.md) for details)._

**Reply by cweagans** [reply]
Wow! That was quick! Thanks for doing that!!
